pkg/router/template/router.go

```diff
@@ -171,13 +172,23 @@ func newTemplateRouter(cfg templateRouterCfg) (*templateRouter, error) {
 }
 
 func endpointsForAlias(alias ServiceAliasConfig, svc ServiceUnit) []Endpoint {
+	// IDhash contains an obfuscated internal IP address that is the value
+	// passed in the cookie. The IP address is made more difficult to extract
+	// by including other internal information in the hash.
 	if len(alias.PreferPort) == 0 {
+		for i := range svc.EndpointTable {
+			endpoint := svc.EndpointTable[i]
```

Comment:
Didn't miss it, misunderstood it. I will make a private function as you 
suggest.
On 04/04/2016 01:54 PM, Ben Bennett wrote:
>
> In pkg/router/template/router.go 
> <https://github.com/openshift/origin/pull/8334#discussion_r58417984>:
>
> >  	if len(alias.PreferPort) == 0 {
> > +		for i := range svc.EndpointTable {
> > +			endpoint := svc.EndpointTable[i]
>
> I think you missed @eparis <https://github.com/eparis>' comment about 
> the go idioms.
>
> It's ugly that we have to do the same thing in two places though. 
> Perhaps make a private function that computes a hash value from an 
> Endpoints structure?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/openshift/origin/pull/8334/files/d5a28c792427c49fed6676cc043d4a13b621d62c#r58417984>
>